third_party/spdk_defs/nvme_defs.h

```diff
@@ -0,0 +1,210 @@
+#ifndef LIB_NVME_DEFS_H
+#define LIB_NVME_DEFS_H
+
+#include <cstdint>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+// https://github.com/spdk/spdk/blob/master/include/spdk/nvme_spec.h
+namespace nvme_defs {
+
+// NVMe Base Specification Figure 125
+// https://nvmexpress.org/wp-content/uploads/NVM-Express-1_4-2019.06.10-Ratified.pdf
+enum class StatusCodeType : uint8_t {
+  kGeneric = 0x0,
+  kCommandSpecific = 0x1,
+  kMediaError = 0x2,
+  kPath = 0x3,
+  // Reserved 0x4-0x6
+  kVendorSpecific = 0x7,
+};
+
+// NVMe Base Specification Figure 126 and Figure 127
+// https://nvmexpress.org/wp-content/uploads/NVM-Express-1_4-2019.06.10-Ratified.pdf
+enum class GenericCommandStatusCode : uint8_t {
+  kSuccess = 0x00,
+  kInvalidOpcode = 0x01,
+  kInvalidField = 0x02,
+  kCommandIdConflict = 0x03,
+  kDataTransferError = 0x04,
+  kAbortedPowerLoss = 0x05,
+  kInternalDeviceError = 0x06,
+  kAbortedByRequest = 0x07,
+  kAbortedSqDeletion = 0x08,
+  kAbortedFailedFused = 0x09,
+  kAbortedMissingFused = 0x0a,
+  kInvalidNamespaceOrFormat = 0x0b,
+  kCommandSequenceError = 0x0c,
+  kInvalidSglSegDescriptor = 0x0d,
+  kInvalidNumSglDescirptors = 0x0e,
+  kDataSglLengthInvalid = 0x0f,
+  kMetadataSglLengthInvalid = 0x10,
+  kSglDescriptorTypeInvalid = 0x11,
+  kInvalidControllerMemBuf = 0x12,
+  kInvalidPrpOffset = 0x13,
+  kAtomicWriteUnitExceeded = 0x14,
+  kOperationDenied = 0x15,
+  kInvalidSglOffset = 0x16,
+  // Reserved 0x17
+  kHostidInconsistentFormat = 0x18,
+  kKeepAliveExpired = 0x19,
+  kKeepAliveInvalid = 0x1a,
+  kAbortedPreempt = 0x1b,
+  kSanitizeFailed = 0x1c,
+  kSanitizeInProgress = 0x1d,
+  kSglDataBlockGranularityInvalid = 0x1e,
+  kCommandInvalidInCmb = 0x1f,
+
```

Comment:
nit: newlines between non-adjacent enum values is not consistent with master

Reply:
Good point!